### ISSUE TITLE
rop search is too slow

### ISSUE THREAD

**Issue by crowell** [radare/radare2#5162]
https://gist.github.com/0c0f34cfcce3953241fae0434baa1f5e

cachegrind output of rop search.

i haven't done any digging yet, but it looks like ~30% is in `r_list_contains` 

probably list is not the right data structure for what's being done here.

I can look deeper this weekend.

**Comment by Maijin** [comment]
Watch out duplicates https://github.com/radare/radare2/issues?utf8=✓&q=rop+searchis%3Aissue+is%3Aopen+

**Comment by radare** [comment]
Sdb 🍆

**Comment by radare** [comment]
ping

**Comment by tkmru** [comment]
It has already been cached using SDB by this commit. [rop.db should work with all /R commands · Issue #5394 · radare/radare2](https://github.com/radare/radare2/issues/5394)

**Comment by alvarofe** [comment]
but the search is made using sdb or once the everything has been found saved in sdb for future reference?

**Comment by radare** [comment]
Yes but the way its stored it doesnt makes it much handy. I would go for enhancing this as main prio and maybe a blogpost too. Anyone?

**Comment by alvarofe** [comment]
The main issue is that search algorithm is not fast not about storing the info, that will be another issue then. A blogpost about what?
